cpp/include/cudf/strings/string_view.inl

```diff
@@ -261,13 +256,15 @@ __device__ inline int string_view::compare(const string_view& in) const
 
 __device__ inline int string_view::compare(const char* data, size_type bytes) const
 {
+    size_type const len1 = size_bytes();
+    if( len1==0 && bytes==0 )
```

Comment:
Is this check needed? 

It looks like if `len1==0` and `bytes==0` it will already fall through to the case of returning 0.

Reply:
Since,  there is no distinction between null and empty anymore in this class the code below this if will incorrectly compare an empty `string_view` created with a nullptr (which is now valid if all the strings in a column happen to be empty) with a `string_view` created with a valid pointer but no size.
This if-statement therefore correctly returns the 0 result when comparing two empty string_views regardless how they are created.

Reply:
Looking at this closer, those following if-statements are no longer needed as well. Just ran full gtests and pytest and verified they are not needed.